lib/validation.js

```diff
@@ -140,7 +150,17 @@ function validate (context, request, execution) {
   }
 
   if (runExecution || !execution.skipBody) {
-    const body = validateParam(context[bodySchema], request, 'body')
+    let validatorFunction = null
+    if (typeof context[bodySchema] === 'function') {
+      validatorFunction = context[bodySchema]
+    } else if (context[bodySchema]) {
+      const mediaName = request.headers['content-type']?.split(';', 1)[0]
```

Comment:
`nit`:

This would be interesting to have them parsed on every request and stored as object.

Reply:
are you suggesting something similar to how it's handled here?

https://github.com/fastify/fastify/blob/f8dbeaae75de1910770890ce5265d82d74f7471f/lib/contentTypeParser.js#L117-L125

Reply:
Agree on it - we should add a `request.contentType` to reuse it. Could you add a `// todo` comment in the code? We can improve it in a following PR 👍🏼